aws-pinpoint-core/src/main/java/com/amplifyframework/pinpoint/core/EventRecorder.kt

```diff
@@ -198,8 +198,11 @@ class EventRecorder(
         return eventIdToDelete
     }
 
-    private fun isRetryableError(code: Int): Boolean {
-        return code in 500..599
+    private fun isRetryableError(code: Int?): Boolean {
+        code?.let {
```

Comment:
`null in 500..599` should be always false. Do we need the null check on `code`?

Reply:
Thats what I am doing with this change, if its a null it will return false. I can make it cleaner by doing an if else. Whether or not this null guard is actually needed or not was out of scope as I simply did the kotlin upgrade and was not auditing the tests.

Reply:
Im saying just `return code in 500..599` would do.

Reply:
ok yeah gotcha. done